src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -1933,5 +1933,81 @@ public interface IOut<out T> { }
                 //         _ = i switch { 1 => default, _ => x }/*T:T*/; // 6
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(46, 29));
         }
+
+        [Fact]
+        [WorkItem(39264, "https://github.com/dotnet/roslyn/issues/39264")]
+        public void IsPatternSplitState_01()
+        {
+            CSharpCompilation c = CreateNullableCompilation(@"
+#nullable enable
+class C
+{
+    string? field = string.Empty;
+
+    void M1(C c)
+    {
+        if (c.field == null) return;
+        
+        c.field.ToString();
+    }
+    
+    void M2(C c)
+    {
+        if (c is { field: null }) return;
```

Comment:
Consider adding a test with an unrelated field, to ensure that we still issue a warning when its accessed.